main/src/main/java/org/mobilitydata/gtfsvalidator/validator/DuplicateRouteNameValidator.java

```diff
@@ -45,50 +46,50 @@
 
   @Override
   public void validate(NoticeContainer noticeContainer) {
-    final Map<String, GtfsRoute> routeByLongName = new HashMap<>(routeTable.entityCount());
-    final Map<String, GtfsRoute> routeByShortName = new HashMap<>(routeTable.entityCount());
-    final Map<String, GtfsRoute> routeByShortAndLongName = new HashMap<>(routeTable.entityCount());
+    final Map<Integer, GtfsRoute> routeByLongName = new HashMap<>(routeTable.entityCount());
+    final Map<Integer, GtfsRoute> routeByShortName = new HashMap<>(routeTable.entityCount());
+    final Map<Integer, GtfsRoute> routeByShortAndLongName = new HashMap<>(routeTable.entityCount());
     routeTable
         .getEntities()
         .forEach(
             route -> {
               if (route.hasRouteShortName() && route.hasRouteLongName()) {
-                if (routeByShortAndLongName.containsKey(getShortAndLongNameKey(route))) {
+                if (routeByShortAndLongName.containsKey(getShortAndLongNameRouteTypeHash(route))) {
```

Comment:
This doesn't check for `areRoutesFromSameAgency()` like the others - should it? It's currently being compared to all other routes, so it seems like it should be. If it needs to change, the tests should be updated with the case that should have failed here.

Reply:
I actually wonder if it would be simpler if we just issued one notice per field as similarly suggested by @aababilov in https://github.com/MobilityData/gtfs-validator/pull/798#discussion_r594776483: if a route has duplicate for both names it has duplicate for route_short_name and has duplicate for route_long_name.

Reply:
☝️  Sounds good to me

Reply:
Done in https://github.com/MobilityData/gtfs-validator/pull/806/commits/c8b90740ba2bd3ba5ceb290863447a38e01fcc1b